packages/zevm-app-contracts/test/instant-rewards/instant-rewards.ts

```diff
@@ -0,0 +1,166 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+import { BigNumber, utils } from "ethers";
+import { ethers } from "hardhat";
+
+import { InstantRewards } from "../../typechain-types";
+import { ClaimData, getSignature } from "./test.helpers";
+
+describe("Instant Rewards Contract test", () => {
+  let instantRewards: InstantRewards,
+    owner: SignerWithAddress,
+    signer: SignerWithAddress,
+    user: SignerWithAddress,
+    addrs: SignerWithAddress[];
+
+  const encodeTaskId = (taskId: string) => utils.keccak256(utils.defaultAbiCoder.encode(["string"], [taskId]));
+
+  const getClaimDataSigned = async (
+    signer: SignerWithAddress,
+    amount: BigNumber,
+    sigExpiration: number,
+    taskId: string,
+    to: string
+  ) => {
+    const claimData: ClaimData = {
+      amount,
+      sigExpiration,
+      taskId,
+      to,
+    };
+
+    const signature = await getSignature(signer, claimData);
+    return {
+      ...claimData,
+      signature,
+    };
+  };
+
+  beforeEach(async () => {
+    [owner, signer, user, ...addrs] = await ethers.getSigners();
+    const instantRewardsFactory = await ethers.getContractFactory("InstantRewards");
+
+    instantRewards = await instantRewardsFactory.deploy(signer.address, owner.address);
+
+    await instantRewards.deployed();
+  });
+
+  it("Should claim", async () => {
+    const currentBlock = await ethers.provider.getBlock("latest");
+    const sigExpiration = currentBlock.timestamp + 1000;
+    const amount = utils.parseEther("1");
+    const taskId = encodeTaskId("WALLET/TASK/EPOC");
+    const to = owner.address;
+
+    // transfer some funds to the contract
+    await owner.sendTransaction({
+      to: instantRewards.address,
+      value: amount,
+    });
+
+    const claimDataSigned = await getClaimDataSigned(signer, amount, sigExpiration, taskId, to);
+
+    const tx = instantRewards.claim(claimDataSigned);
+    await expect(tx).to.emit(instantRewards, "Claimed").withArgs(owner.address, taskId, amount);
+  });
+
+  it("Should claim if pause and unpause", async () => {
+    const currentBlock = await ethers.provider.getBlock("latest");
+    const sigExpiration = currentBlock.timestamp + 1000;
+    const amount = utils.parseEther("1");
+    const taskId = encodeTaskId("WALLET/TASK/EPOC");
+    const to = owner.address;
+
+    await instantRewards.pause();
+    await instantRewards.unpause();
+
+    // transfer some funds to the contract
+    await owner.sendTransaction({
+      to: instantRewards.address,
+      value: amount,
+    });
+
+    const claimDataSigned = await getClaimDataSigned(signer, amount, sigExpiration, taskId, to);
+
+    const tx = instantRewards.claim(claimDataSigned);
+    await expect(tx).to.emit(instantRewards, "Claimed").withArgs(owner.address, taskId, amount);
+  });
+
+  it("Should revert if try to claim on behalf of somebody else", async () => {
```

Comment:
I think we're missing one scenario to test: `Should revert if try to claim twice`?

Reply:
added